### PR TITLE
Search component

### PR DESCRIPTION
**Pull request by marySalvi**
Components Added:

- Search
- Predicate
- GeoJsonForm

Expected Functionality:
The Search component pulls in the predicate, geoJsonForm and dateRange components. With Storybook you should be able to interact with all the components however clicking search does (and should) throw an error since it makes an API call.

### CODE REVIEW

File: src/store/cesium/search.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+          ),
+          Cesium.Math.toDegrees(
+            (Cesium.Cartographic.fromCartesian(element)
+            ).latitude,
+          ),
+        ]]);
+      });
+      polyPoints.push(polyPoints[0]);
+      drawnShape.value.type = 'Polygon';
+      // This is super duper important!!!!
```
Comment (mvandenburgh) [inline src/store/cesium/search.ts:107]: Can this comment be expanded a little to explain why it's important? It's not immediately clear to me from looking at it _why_ it's super duper important :smile:

File: src/store/types.ts [hunk 2]
```diff
@@ -11,23 +11,49 @@ export type Footprint = {
...
-      end: string | null;
-    };
-    time: {
-      startTime: string | null;
-      endTime: string | null;
-    };
-  }
+  id: number | null;
+  name: string | null;
+}export interface ResultsFilter {
```
Comment (mvandenburgh) [inline src/store/types.ts:44]:
```suggestion
}

export interface ResultsFilter {
```

File: src/components/molecules/GeoJsonForm.vue [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+    const isGeoJSON = (inputText: string) => {
+      try {
+        JSON.parse(inputText);
+        geoJsonErrorMessages.value = [];
+      } catch (e) {
+        geoJsonErrorMessages.value = ['Not a valid GeoJSON Polygon or MultiPolygon'];
+      }
+      return true;
+    };
```
Comment (mvandenburgh) [inline src/components/molecules/GeoJsonForm.vue:41]: This might just be a personal preference on my part, but when I see a function named like `isXXX`, I expect it to be a pure function that simply checks if something is true. This function does that, but also has the side effect of populating the `geoJsonErrorMessages` list.

Also, the return type should probably be `boolean` in all cases -
```suggestion
    const isGeoJSON = (inputText: string) => {
      try {
        JSON.parse(inputText);
        geoJsonErrorMessages.value = [];
      } catch (e) {
        geoJsonErrorMessages.value = ['Not a valid GeoJSON Polygon or MultiPolygon'];
        return false;
      }
      return true;
    };
```

File: src/store/search.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+});
+
+export const updateResults = async () => {
+  const collectionIDs: number[] = [];
+  // eslint-disable-next-line no-unused-expressions
+  resultsFilter.value.collections?.forEach((element) => {
+    if (element.id) {
+      collectionIDs.push(element.id);
+    }
+  });
```
Comment (mvandenburgh) [inline src/store/search.ts:60]: I believe this can be simplified to
```suggestion
  const collectionIDs = resultsFilter.value.collections?.filter((element) => element.id).map((element) => element.id);
```

File: src/components/molecules/GeoJsonForm.vue [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+  setup() {
+
+    const geoOptions = [
+      'Draw on Map',
+      'Copy/Paste GeoJson',
+      'Upload File',
+    ];
+    const geoJsonString = ref('');
+    const geoJsonErrorMessages = ref(['']);
```
Comment (mvandenburgh) [inline src/components/molecules/GeoJsonForm.vue:19]:
```suggestion
    const geoJsonErrorMessages = ref([]);
```

Does other code depend on this starting off with a single empty string as an element?

**Review by mvandenburgh — CHANGES_REQUESTED** [review]

File: src/store/search.ts (hunk 4 above)
Comment (waxlamp) [reply src/store/search.ts:60]: Unless `element.id` has a numeric type, in which case you might need a more specific test (against `null` or `undefined`, etc.).
